spec/datagrid/filters_spec.rb

```diff
@@ -292,5 +292,33 @@ class TestGrid8728
       expect(non_admin_filters).to_not include(:id)
       expect(non_admin_filters).to include(:name)
     end
+
+    context 'with delegation to attribute' do
+      let(:role) { OpenStruct.new('admin?' => admin) }
+      let(:klass) do
+        test_report_class do
+          attr_accessor :role
+          delegate :admin?, to: :role
```

Comment:
The case without the delegation is basically the same test as the one already in place for the condition. Which works because there is a filter defined for _admin_mode_, it fails otherwise: 

```
  1) Datagrid::Filters :if :unless options supports :if option
     Failure/Error: self.send(:"#{attribute}=", value)

     NoMethodError:
       undefined method `admin_mode=' for #<#<Class:0x007fd320f98760>:0x007fd320f694b0>
     # ./lib/datagrid/core.rb:96:in `[]='
     # ./lib/datagrid/core.rb:108:in `block in assign_attributes'
     # ./lib/datagrid/core.rb:107:in `each'
     # ./lib/datagrid/core.rb:107:in `assign_attributes'
     # ./lib/datagrid/core.rb:72:in `initialize'
```